camp/camp-base/src/main/java/org/apache/brooklyn/camp/spi/collection/BasicResourceLookup.java

```diff
@@ -43,6 +47,11 @@ public synchronized void add(T item) {
         if (old!=null) {
             items.put(old.getId(), old);
             throw new IllegalStateException("Already contains item for "+item.getId()+": "+old+" (adding "+item+")");
```

Comment:
agree doesn't make much sense -- have added a TODO comment (as wasn't introduced here and don't want to change, even though i'm pretty sure it's safe)